### ISSUE TITLE
make path handling consistent

### ISSUE THREAD

**Issue by rwcarlsen** [canonical/operator#742]
We have a mixture of pathlib and string paths variously around the framework.  This should be made consistent everywhere - probably just using pathlib for all the file paths we can.

**Comment by benhoyt** [comment]
I agree we should be consistent for new APIs (though I tend to prefer ordinary strings). However, I'm not sure it's worth trying to change existing APIs -- closing.
